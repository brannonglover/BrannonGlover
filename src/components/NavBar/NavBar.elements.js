import styled from 'styled-components/macro';

export const NavBarWrapper = styled.div`
  width: 100%;
  background-color: #414047;
  border-bottom: 2px solid #a83214;
  position: fixed;
  top: 0;
  z-index: 5;
`;

export const NavBarContainer = styled.div`
  max-width: 75em;
  display: flex;
  margin: 0 auto;
`;

export const Avatar = styled.div`
  display: flex;
  flex: 1;
  align-items: center;
`;

export const Image = styled.div`
  width: 3em;
  height: auto;
  padding: .5em;

  > img {
    width: 100%;
    height: auto;
    display: block;
    border-radius: 50%;
    border: 1px solid #fff;
  }
`;

export const Title = styled.div`
  font-size: 1.2em;
  color: #fff;
  padding-left: .5em;
`;

export const Hamburger = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-right: 1em;

  @media screen and (min-width: 768px) {
    display: none;
  }

  div {
    width: 2em;
    height: .25em;
    background-color: #fff;
    margin: .15em 0;
    border-radius: 20px;
  }
`;

export const Links = styled.div`
  display: none;

  @media screen and (min-width: 768px) {
    display: flex;

    a {
      text-decoration: none;
      color: #fff;
    }
  }
`;