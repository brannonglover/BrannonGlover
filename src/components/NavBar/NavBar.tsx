import React from "react";
import {
  NavBarWrapper,
  NavBarContainer,
  Avatar,
  Image,
  Title,
  Hamburger,
} from "./NavBar.elements";
import Me from "../../assets/images/me.png";
import strings from "../../common/strings";
import Menu from "../Menu/Menu";

interface INavBarProps {
  menuToggle: () => void;
  ref: React.Ref<HTMLDivElement>;
}

const NavBar: React.FC<INavBarProps> = React.forwardRef(
  ({ menuToggle }, ref: React.Ref<HTMLDivElement>) => {
    return (
      <NavBarWrapper>
        <NavBarContainer ref={ref}>
          <Avatar>
            <Image>
              <img src={Me} alt="Brannon Glover" />
            </Image>
            <Title>{strings.WEB_DEV}</Title>
          </Avatar>
          <Menu />
          <Hamburger onClick={menuToggle}>
            <div></div>
            <div></div>
            <div></div>
          </Hamburger>
        </NavBarContainer>
      </NavBarWrapper>
    );
  }
);

export default NavBar;
