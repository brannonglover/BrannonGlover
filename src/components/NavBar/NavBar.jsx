import React from "react";
import { NavBarContainer, Avatar, Links, Hamburger } from "./NavBar.elements";
import Me from "../../assets/images/me.png";

export default function NavBar() {
  return (
    <NavBarContainer>
      <Avatar>
        <img src={Me} alt="Brannon Glover" />
      </Avatar>
      <Hamburger>
        <div></div>
        <div></div>
        <div></div>
      </Hamburger>
      <Links>
        <a
          href="https://github.com/brannonglover"
          rel="noreferrer"
          target="_blank"
        >
          GitHub
        </a>
        <a href="https://twitter.com/BrannonGlover" rel="noreferrer">
          Twitter
        </a>
      </Links>
    </NavBarContainer>
  );
}
