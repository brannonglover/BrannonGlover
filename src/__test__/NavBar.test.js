import React from 'react';
import { render } from '@testing-library/react';
import NavBar from '../components/NavBar/NavBar';

test('Render NavBar component', () => {
  const { asFragment } = render(<NavBar />);

  expect(asFragment()).toMatchSnapshot();
});