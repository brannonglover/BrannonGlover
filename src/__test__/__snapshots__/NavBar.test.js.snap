// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NavBar component 1`] = `
<DocumentFragment>
  <div
    class="NavBarelements__NavBarWrapper-sc-15n5tzl-0 hDaOpt"
  >
    <div
      class="NavBarelements__NavBarContainer-sc-15n5tzl-1 dAleMO"
    >
      <div
        class="NavBarelements__Avatar-sc-15n5tzl-2 dpvuOI"
      >
        <div
          class="NavBarelements__Image-sc-15n5tzl-3 fWapQl"
        >
          <img
            alt="Brannon Glover"
            src="me.png"
          />
        </div>
        <div
          class="NavBarelements__Title-sc-15n5tzl-4 hDYNLT"
        >
          Web Developer
        </div>
      </div>
      <div
        class="Menuelements__MenuWrapper-k5kaq1-0 httLqd"
      >
        <ul>
          <li>
            <a
              href="https://github.com/brannonglover?tab=repositories"
              rel="noreferrer"
              target="_blank"
            >
              GitHub
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/BrannonGlover"
              rel="noreferrer"
              target="_blank"
            >
              Twitter
            </a>
          </li>
          <li>
            <a
              href="https://medium.com/@brannonglover"
              rel="noreferrer"
              target="_blank"
            >
              Medium
            </a>
          </li>
          <li>
            <a
              href="https://www.linkedin.com/in/brannon-glover-31a22212/"
              rel="noreferrer"
              target="_blank"
            >
              LinkedIn
            </a>
          </li>
        </ul>
      </div>
      <div
        class="NavBarelements__Hamburger-sc-15n5tzl-5 gfUypB"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
